File: 2ºCARRERA/tprog/practica2_872815_873338/c++/push.cc
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : push.cc
 */

#include "push.h"

Push::Push(int32_t c)
    :num(c)
{}
void Push::ejecuta(stack<int32_t> &pila, int& pc) {
    pila.push(num);
    pc++;
}

void Push::muestra(){
    cout << "push " << num;
}

File: 2ºCARRERA/tprog/practica2_872815_873338/java/Push.java
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : Push.java
 */

import java.util.Stack;

class Push implements Instruccion
{
    private int num;
    public Push(int n) {
        num = n;
    }
    public int ejecuta(Stack<Integer> pila, int pc) {
       pila.push(num);
       pc++;
       return pc;
    }
    public void muestra() {
        System.out.print("push " + num);
    }
}